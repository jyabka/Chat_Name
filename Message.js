import React from 'react';

export default class Message extends React.Component {
    render() {
        const { nick, message } = this.props;
        return <li>
            <b>{nick}:</b>
            {message}
        </li>;
    }
}
