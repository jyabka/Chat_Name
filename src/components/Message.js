import React from 'react';

class Message extends React.Component {
    render() {
        const { nickname, content } = this.props;
        return (
            <li>
                <b>{nickname}:</b>
                {content}
            </li>
        );
    }
}

export default Message;
