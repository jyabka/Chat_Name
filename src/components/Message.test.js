import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Message from './Message';

describe('Message component', () => {
    test('Message matches snapshot', () => {
        const nick = 'test';
        const message = 'test';
        const component = renderer.create(<Message nick={nick} message={message} />).toJSON();
        expect(component).toMatchSnapshot();
    });

    test('Message shows nick and message', () => {
        const nick = 'test';
        const message = 'test';
        const component = shallow(<Message nick={nick} message={message} />);
        expect(component.text()).toContain(nick);
        expect(component.text()).toContain(message);
    });
});
