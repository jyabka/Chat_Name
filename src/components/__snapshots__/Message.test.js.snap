// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component Message matches snapshot 1`] = `
<li>
  <b>
    test
    :
  </b>
  test
</li>
`;
